File: leetcode/ConcatenateArray.java
package leetcode;

public class ConcatenateArray {
    public int[] concatenateArray(int[] nums) {
        int n = nums.length;
        int[] newArray = new int[2 * n];

        for(int i = 0; i < n; i++) {
            newArray[i] = newArray[i + n] = nums[i];
        }
    }
}

File: leetcode/concatenate-array.go
package main

//https://leetcode.com/problems/concatenation-of-array/

//Creates a new array to hold concatenated values. Consumes more memory.
func getConcatenation1(nums []int) []int {
	ans := []int{}

	for range 2 {
		for _, v := range nums {
			ans = append(ans, v)
		}
	}
	return ans
}

//Modifies the input array in place
func getConcatenation2(nums []int) []int {
	length := len(nums)

	for i := 0; i < length; i++ {
		nums = append(nums, nums[i])
	}
	return nums
}
